lib/text/lrc_text_parser.js

```diff
@@ -95,27 +98,57 @@ shaka.text.LrcTextParser = class {
     const LrcTextParser = shaka.text.LrcTextParser;
     const match = LrcTextParser.timeFormat_.exec(string);
     const minutes = parseInt(match[1], 10);
-    const seconds = parseInt(match[2], 10);
-    const hundredthsOfSeconds = match[4] ? parseInt(match[4], 10) : 0;
-    return minutes * 60 + seconds + hundredthsOfSeconds / 100;
+    const seconds = parseFloat(match[2]);
+    return minutes * 60 + seconds;
+  }
+
+  /**
+   * Convert a LRC input to WebVTT
+   *
+   * @param {BufferSource} data
+   * @param {number} segmentEnd
+   * @return {string}
+   * @export
+   */
+  static convertToWebVTT(data, segmentEnd) {
+    const LrcTextParser = shaka.text.LrcTextParser;
+    const cues = LrcTextParser.getCues_(data, segmentEnd);
+    let webvttString = 'WEBVTT\n\n';
+    for (const cue of cues) {
+      const webvttTimeString = (time) => {
+        const hours = Math.floor(time / 3600);
+        const minutes = Math.floor(time / 60 % 60);
+        const seconds = Math.floor(time % 60);
+        const milliseconds = Math.floor(time * 1000 % 1000);
+        return (hours < 10 ? '0' : '') + hours + ':' +
+            (minutes < 10 ? '0' : '') + minutes + ':' +
+            (seconds < 10 ? '0' : '') + seconds + '.' +
+            (milliseconds < 100 ? (milliseconds < 10 ? '00' : '0') : '') +
+            milliseconds;
+      };
+      webvttString += webvttTimeString(cue.startTime) + ' --> ' +
+          webvttTimeString(cue.endTime) + '\n';
+      webvttString += cue.payload + '\n\n';
+    }
+    return webvttString;
   }
 };
 
 /**
  * @const
  * @private {!RegExp}
- * @example 50t or 50.5t
+ * @example [mm:ss.xx]Text
```

Comment:
Also accepts commas and 1-3 digits past the decimal

Reply:
Done!